edgexfoundry/device-rest-go
# feat: replace gorilla/mux with labstack/echo

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rest-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

Run make docker to build the image
From the latest compose-builder run `make run no-secty device-dev ds-rest`
Set the log level of device-rest-go to `DEBUG`
Verify the device-rest-go service bootstraps successfully and contains the following log message:
```
level=INFO ts=2023-08-28T08:18:13.180389969Z app=device-rest source=resthandler.go:65 msg="Route /api/v3/resource/:deviceName/:resourceName added."
```
Execute the following command:
```
curl -X POST localhost:59986/api/v3/resource/test/test
```
Verify the log contains the following message:
```
level=DEBUG ts=2023-08-28T08:19:52.992276584Z app=device-rest source=resthandler.go:83 msg="Received POST for Device=test Resource=test"
```

## New Dependency Instructions (If applicable)
